### DISCUSSION TITLE
Testing components using injection tokens (multi)

### DISCUSSION THREAD

**Discussion by LukasMachetanz** [help-me-mom/ng-mocks#5537] [Q&A] [answered]
Hey, is it possible to mock a specific `InjectionToken` which is assumed to be provided using the `multi` flag within a dedicated test, e.g. using `MockInstance`?

**Comment by LukasMachetanz** [comment]
Currently, I am doing it like this:

```
    MockInstance(DISCOVER_CONTRIBUTION, () => {
      return [
        {
          id: "one",
          title: "One",
          icon: "sds-icon--one"
        },
        {
          id: "two",
          title: "Two",
          icon: "sds-icon--two",
          shouldApply: (): boolean => true
        },
        {
          id: "three",
          title: "Three",
          icon: "sds-icon--three",
          shouldApply: (): boolean => false
        }
      ] as never;
    });
```

But I would like to prevent using the cast `as never`. Is this somehow possible?

**Reply by satanTime** [reply]
I think it should be a feature request: https://github.com/help-me-mom/ng-mocks/issues/5585

**Reply by satanTime** [reply, marked as answer]
A fix has been released in v.14.10.1: https://ng-mocks.sudo.eu/api/MockInstance#customizing-tokens
